bugsnag-android-core/src/main/java/com/bugsnag/android/ThreadState.kt

```diff
@@ -5,37 +5,29 @@ import java.io.IOException
 /**
  * Capture and serialize the state of all threads at the time of an exception.
  */
-internal class ThreadState : JsonStream.Streamable {
-
-    private val threads: List<Thread>
-
-    constructor(threads: List<Thread>) {
-        this.threads = threads
-    }
-
-    constructor(
-        config: ImmutableConfig,
-        currentThread: java.lang.Thread,
-        stackTraces: MutableMap<java.lang.Thread, Array<StackTraceElement>>,
-        exc: Throwable?
-    ) {
+internal class ThreadState @JvmOverloads constructor
+    (
+    config: ImmutableConfig,
+    excStacktrace: Array<StackTraceElement>,
+    currentThread: java.lang.Thread = java.lang.Thread.currentThread(),
+    stackTraces: MutableMap<java.lang.Thread, Array<StackTraceElement>> = java.lang.Thread.getAllStackTraces()
```

Comment:
Does this need to be a parameter? Can't it just be a local in init?

Reply:
I believe the intention behind this would be to make testing easier. However, as we're not currently using this any test code I can move this out of the constructor.